src/plugins/chart_expressions/expression_gauge/public/components/utils/use_gauge_size_by_type.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useEffect } from 'react';
+import { ChartSizeSpec } from '@kbn/chart-expressions-common';
+import { GaugeShape, GaugeShapes } from '../../../common';
+
+export const useGaugeSizeByType = (type: GaugeShape, setChartSize: (d: ChartSizeSpec) => void) => {
+  useEffect(() => {
+    const maxDimensions: Record<GaugeShape, ChartSizeSpec> = {
```

Comment:
Personally, I felt the 600x600 maximum was fine both on my ultrawide and laptop monitors during testing, but I'll defer to @gvnmagni if he feels otherwise.